src/components/main-track-editor.js

```diff
@@ -0,0 +1,227 @@
+        <Input value={sourceUrl} onChange={handleSourceUrlChange} onBlur={handleSourceUrlBlur} />
+      </FormItem>
+      )}
+      {sourceType === MEDIA_SOURCE_TYPE.internal && (
+      <FormItem label={t('common:internalUrl')} {...formItemLayout}>
+        <div className="u-input-and-button">
+          <Input
+            addonBefore={CDN_URL_PREFIX}
+            value={sourceUrl}
+            onChange={handleSourceUrlChange}
+            onBlur={handleSourceUrlBlur}
+            />
+          <ResourcePicker
+            url={storageLocationPathToUrl(sourceUrl)}
+            onUrlChange={url => handleSourceUrlChangeComplete(urlToStorageLocationPath(url))}
+            />
+        </div>
+      </FormItem>
+      )}
+      {sourceType === MEDIA_SOURCE_TYPE.youtube && (
+      <FormItem label={t('common:youtubeUrl')} {...formItemLayout} {...validation.validateUrl(sourceUrl, t)} hasFeedback>
+        <Input value={sourceUrl} onChange={handleSourceUrlChange} onBlur={handleSourceUrlBlur} />
+      </FormItem>
+      )}
+      <FormItem label={t('common:aspectRatio')} {...formItemLayout}>
+        <RadioGroup
+          size="small"
+          defaultValue={MEDIA_ASPECT_RATIO.sixteenToNine}
+          value={aspectRatio}
+          onChange={handleAspectRatioChanged}
+          disabled={![RESOURCE_TYPE.video, RESOURCE_TYPE.none].includes(getResourceType(sourceUrl))}
+          >
+          {Object.values(MEDIA_ASPECT_RATIO).map(ratio => (
+            <RadioButton key={ratio} value={ratio}>{ratio}</RadioButton>
+          ))}
+        </RadioGroup>
+      </FormItem>
+      <FormItem label={t('common:videoDisplay')} {...formItemLayout}>
+        <Switch
+          size="small"
+          checked={showVideo}
+          onChange={handleShowVideoChanged}
+          disabled={![RESOURCE_TYPE.video, RESOURCE_TYPE.none].includes(getResourceType(sourceUrl))}
+          />
+      </FormItem>
+      <FormItem label={t('playbackRange')} {...formItemLayout}>
+        <div className="u-input-and-button">
+          <Input value={renderPlaybackRangeInfo()} readOnly />
+          <MediaRangeSelector
+            range={playbackRange}
+            onRangeChange={handlePlaybackRangeChange}
+            sourceUrl={getFullSourceUrl({ url: sourceUrl, sourceType, cdnRootUrl: clientConfig.cdnRootUrl })}
+            />
+        </div>
+      </FormItem>
+      <FormItem label={t('common:copyrightNotice')} {...formItemLayout}>
+        <MarkdownInput value={copyrightNotice} onChange={handleCopyrightNoticeChanged} />
+      </FormItem>
+    </Fragment>
+  );
+}
+
+MainTrackEditor.propTypes = {
+  content: PropTypes.shape({
+    sourceType: PropTypes.oneOf(Object.values(MEDIA_SOURCE_TYPE)),
+    sourceUrl: PropTypes.string,
+    showVideo: PropTypes.bool,
+    aspectRatio: PropTypes.oneOf(Object.values(MEDIA_ASPECT_RATIO)),
+    playbackRange: PropTypes.arrayOf(PropTypes.number),
+    copyrightNotice: PropTypes.string
+
```

Comment:
You know...

Reply:
Gone like a bad dream when morning strikes..